src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -12,8 +12,8 @@ public static class Enumerable
     {
         public static IEnumerable<TSource> Where<TSource>(this IEnumerable<TSource> source, Func<TSource, bool> predicate)
         {
-            if (source == null) throw Error.ArgumentNull("source");
-            if (predicate == null) throw Error.ArgumentNull("predicate");
+            Error.CheckSourceNotNull(source);
+            Error.CheckPredicateNotNull(predicate);
```

Comment:
With small enough methods a reason for doing it can be precisely to encourage inlining (if the result is that the containing method drops below the jitter's threshold). I think you've a good point about it being in a different PR though. I shall revert this particular commit soon.